devp2p/peer.py

```diff
@@ -247,10 +253,6 @@ def _run_ingress_message(self):
                     log.debug('multiplexer error', peer=self, error=e)
                     self.report_error('multiplexer error')
                     self.stop()
-            else:
-                log.debug('no data on socket', peer=self)
-                self.report_error('no data on socket')
-                self.stop()
 
```

Comment:
Btw, I kept getting this error. Not sure about gevent, but async sockets can become ready for reading even without any pending data.